exchanges/bybit/bybit_websocket.go

```diff
@@ -0,0 +1,453 @@
+		err := json.Unmarshal(respRaw, &data)
+		if err != nil {
+			return err
+		}
+
+		p, err := by.extractCurrencyPair(data.Parameters.Symbol, asset.Spot)
+		if err != nil {
+			return err
+		}
+
+		side := order.Sell
+		if data.TradeData.Side {
+			side = order.Buy
+		}
+
+		return trade.AddTradesToBuffer(by.Name, trade.Data{
+			Timestamp:    data.TradeData.Time.Time(),
+			CurrencyPair: p,
+			AssetType:    asset.Spot,
+			Exchange:     by.Name,
+			Price:        data.TradeData.Price,
+			Amount:       data.TradeData.Size,
+			Side:         side,
+			TID:          data.TradeData.ID,
+		})
+
+	case wsTicker:
+		var data WsSpotTicker
+		err := json.Unmarshal(respRaw, &data)
+		if err != nil {
+			return err
+		}
+
+		p, err := by.extractCurrencyPair(data.Ticker.Symbol, asset.Spot)
+		if err != nil {
+			return err
+		}
+
+		by.Websocket.DataHandler <- &ticker.Price{
+			ExchangeName: by.Name,
+			Bid:          data.Ticker.Bid,
+			Ask:          data.Ticker.Ask,
+			LastUpdated:  data.Ticker.Time.Time(),
+			AssetType:    asset.Spot,
+			Pair:         p,
+		}
+
+	case wsMarkets:
+		var data KlineStream
+		err := json.Unmarshal(respRaw, &data)
+		if err != nil {
+			return err
+		}
+
+		p, err := by.extractCurrencyPair(data.Kline.Symbol, asset.Spot)
+		if err != nil {
+			return err
+		}
+
+		by.Websocket.DataHandler <- stream.KlineData{
+			Pair:       p,
+			AssetType:  asset.Spot,
+			Exchange:   by.Name,
+			StartTime:  data.Kline.StartTime.Time(),
+			Interval:   data.Parameters.KlineType,
+			OpenPrice:  data.Kline.OpenPrice,
+			ClosePrice: data.Kline.ClosePrice,
+			HighPrice:  data.Kline.HighPrice,
+			LowPrice:   data.Kline.LowPrice,
+			Volume:     data.Kline.Volume,
+		}
+
+	default:
+		by.Websocket.DataHandler <- stream.UnhandledMessageWarning{Message: by.Name + stream.UnhandledMessage + string(respRaw)}
+	}
+
+	return nil
+}
+
+func (by *Bybit) wsUpdateOrderbook(update *WsOrderbookData, p currency.Pair, assetType asset.Item) error {
+	if update == nil || (len(update.Asks) == 0 && len(update.Bids) == 0) {
+		return errors.New("no orderbook data")
+	}
+	asks := make([]orderbook.Item, len(update.Asks))
+	for i := range update.Asks {
+		target, err := strconv.ParseFloat(update.Asks[i][0], 64)
+		if err != nil {
+			by.Websocket.DataHandler <- err
+			continue
+		}
+		amount, err := strconv.ParseFloat(update.Asks[i][1], 64)
+		if err != nil {
+			by.Websocket.DataHandler <- err
+			continue
+		}
+		asks[i] = orderbook.Item{Price: target, Amount: amount}
+	}
+	bids := make([]orderbook.Item, len(update.Bids))
+	for i := range update.Bids {
+		target, err := strconv.ParseFloat(update.Bids[i][0], 64)
+		if err != nil {
+			by.Websocket.DataHandler <- err
+			continue
+		}
+		amount, err := strconv.ParseFloat(update.Bids[i][1], 64)
+		if err != nil {
+			by.Websocket.DataHandler <- err
+			continue
+		}
+		bids[i] = orderbook.Item{Price: target, Amount: amount}
+	}
+	return by.Websocket.Orderbook.LoadSnapshot(&orderbook.Base{
```

Comment:
You mean this right?
```
return by.Websocket.Orderbook.Update(&buffer.Update{...})
```

Reply:
Keep it as is, for now, I was only mentioning because of the verbosity it creates. Using [loadsnapshot_see_here](https://github.com/thrasher-corp/gocryptotrader/blob/master/exchanges/stream/buffer/buffer.go#L337) on every update will deploy the orderbook to the websocket routine manager in engine and log info. Wheres [Update_see_here](https://github.com/thrasher-corp/gocryptotrader/blob/master/exchanges/stream/buffer/buffer.go#L167) It will only alert for the sync manager timeout requirements.